### PR TITLE
feat: Implement Application.RequiresPointerMode

### LINKED ISSUE #4371: Implement Application.RequiresPointerMode
## What would you like to be added:

This property is relevant to Xbox only, but it currently throws in Uno.

## Why is this needed:

UWP parity.

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #4371

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

Throws.

## What is the new behavior?

Does not throw.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Application.cs [hunk 1]
```diff
@@ -65,6 +65,8 @@ public static class TraceProvider
 
 		public DebugSettings DebugSettings { get; } = new DebugSettings();
 
+		public ApplicationRequiresPointerMode RequiresPointerMode { get; set; } = ApplicationRequiresPointerMode.Auto;
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Application.cs:68]: It still is not implemented, we probably should raise a message when setting the property.

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Application.cs:68]: Setting the property has effect only on Xbox, it is ignored on other Windows SKUs (https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.application.requirespointermode?view=winrt-19041 in remarks). So we can just have it there and the fact that it does nothing is "expected".

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Application.cs:68]: Ah well, if the docs say so ;)

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Application.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Application.cs:68]: I have used the property before on desktop, so can confirm it does nothing :-D .

**Review by davidjohnoliver — APPROVED** [review]

**Pull request merged**